app/src/main/java/io/gnosis/safe/ui/settings/owner/OwnerSeedPhraseViewModel.kt

```diff
@@ -12,7 +12,42 @@ class OwnerSeedPhraseViewModel
     appDispatchers: AppDispatchers
 ) : PublishViewModel<ImportOwnerKeyState>(appDispatchers) {
 
-    fun validate(seedPhrase: String) {
+    fun validate(seedPhraseOrKey: String) {
+        if (detectPrivateKey(seedPhraseOrKey)) {
+            validatePrivateKey(seedPhraseOrKey)
+        } else {
+            validateSeedPhrase(seedPhraseOrKey)
+        }
+    }
+
+    internal fun detectPrivateKey(seedPhraseOrKey: String): Boolean {
+        val input = removeHexPrefix(seedPhraseOrKey)
+
+        val pattern = "[0-9a-fA-F]{64}".toRegex()
+        pattern.find(input)?.let {
+            return true
+        }
+
+        return false
+    }
+
+    internal fun validatePrivateKey(key: String) {
```

Comment:
there is no validation here except for whether it is a zero key. You should also check if it's a valid hex number. Something like this 0[xX][0-9a-fA-F]{64}

Reply:
Ok. But then we do this check twice. But you are right. When we use this at some point in another part of the application a future versio of us might assume it really validates.